app/src/test/java/org/mozilla/focus/ext/UriKtTest.kt

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.ext
+
+import android.net.Uri
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class UriKtTest {
+
+    @Test
+    fun `WHEN truncatedHost is called on a null host THEN it returns the host`() {
+        val expectedHost = null
+        val uri = httpsUriBuilder()
+            .path("mozilla/en-US")
+            .authority(expectedHost)
+            .build()
+        assertNull(uri.host)
+
+        assertEquals(expectedHost, uri.truncatedHost())
```

Comment:
I'm not sure I see the value of duplicating this 'not null' assertion.

Reply:
@Baron-Severin I don't understand – where is there a not null assertion and where is it duplicated?

Reply:
Sorry, poorly phrased.  I would get rid of the `expectedHost` variable and rewrite as follows:

```kotlin
val uri = httpsUriBuilder()
    .path("mozilla/en-US")
    .authority(null)
    .build()

assertNull(uri.host)
assertNull(uri.truncatedHost())
```

Reply:
Got it, will change. fwiw, the first `assertNull` is part of the WHEN statement – when I construct a Uri using the builder, it's a sanity check that the host is actually null – and the second `assertNull` is part of the THEN. Hence I group the first with the `uri` declaration and the second on its own.